index.rst

```diff
@@ -13,17 +13,188 @@
+
+Service-specific checks
+-----------------------
+
+**Before:** Mobu can be configured to run all notebooks in a given directory and branch, but beyond that all notebook payloads in that directory/branch must work.
+
+**After:** By leveraging a naming convention that matches the application name in phalanx, mobu uses the ArgoCD UI or the environment manifest to only run notebooks corresponding to deployed services.
+
+**Discussion:** We want a more fine-grained way of running only notebooks that are expected to complete successfully.
+However an opt-in approach of creating manifests of specific notebooks in a repo (or a proxy of the same result, such as having mobu run off different branches in different environments) is a maintenance headache.
+If instead we adopt a naming convention for notebooks that match the ArgoCD application name, mobu can be modified to run all notebooks from participating services.
+For example, if at the La Serena base deployment nublado is set to true in the environment yaml and tap is set to false and there were notebooks in system-test named
+
+.. code-block:: bash
+
+      nublado_login.ipynb
+      nublado_dask_cluster.ipynb
+      tap.ipynb
+
+mobu would run the nublado notebooks but not attempt the tap one.
+
+The advantage of this approach is that developers can check in new notebooks for services without necessitating mobu changes.
+Note that currently mobu caches notebooks on start-up and needs to be restarted to pick up new notebooks; an API call or other method to induce a running mobu to re-poll its notebook repos would be useful (but not critical).
+
+Reliance on specific data holdings
+----------------------------------
+
+**Before:** System-test notebooks address specific data holdings
+
+**After:** Notebooks perform a data discovery step and run on arbitrary holdings and/or opt out of data-holding specific checks.
+
+**Discussion:**
+
+From the beginning we have identified the need to have a small data-set that is available on all deployments to allow system-test notebooks to run everywhere.
+While there is merit to this idea, in practice finding the effort to curate such a careful minimal in size but maximal in utility dataset has been hard to find.
+With the advent of the tutorial-notebooks repo, the requirement for performing substantive computations and/or service load has been eliminated from system-test.
+With the proposed data curation notebooks, that require specific data holdings can live elsewhere.
+To the extent that this is practical, system-test service notebooks should be written with a data discovery or data check step to see what data is available (eg. in terms of available catalogs, tap_schema could be queried first to make sure unavailable catalogs are not being requested).
+In practice this is may be an aspirational requirement.
```

Comment:
Note that this means we're adding two different mechanisms to decide whether to skip a notebook: one that mobu does, by checking against the service list, and one that the notebook does, by doing data discovery. That may be fine, but it's a shame that we have two separate mechanisms and thus have to check in two places for whether a notebook will actually run.

The model where we skip the notebook entirely seems conceptually somewhat nicer than the model where we run the notebook but it decides to do nothing, since the latter is less obvious to a human observer.

Reply:
I concur this does imply two mechanism, but I think that if we implement service discovery using a VO registry, that won't really cover non-VO applications (eg portal, plot-navigator etc)?  So I think we still need configuration-driven service "discovery"?  We also want a system that doesn't _require_ you to use a VO registry?

Reply:
I was assuming that we would register all of our services in the VO registry, creating extensions if needed, because running multiple discovery services seems obviously wrong. I suppose the easy way to do that is to provide both a JSON and an XML registry (and I guess we have to provide RegTAP as well), and include our local stuff only in the JSON version so that we don't have to write XML schemas.

I'm very confused by you saying that you want a system that doesn't require you to use a VO registry, mostly around what you mean by "system". mobu itself can just run all the notebooks, run only notebooks in specific directories, etc.; you don't have to provide a registry. We have a mechanism for selecting which notebooks to run based on a VO registry, which you don't have to use. If people want to add a different way of selecting notebooks to run, they can propose it in an issue and we can see how hard it would be to maintain.

Reply:
[following discussion] I had something more simple in mind here (that mobu notebook payloads) could do some probing of a service to find something useful to do with the available srvices, but the point that mobu leverage any future discovery mechanism to decide what to run on what data is well made.

Reply:
Clarified this is not asking for another discovery service
https://github.com/lsst-sqre/sqr-080/commit/118025281805917ced951a6ddb45ce0a62ab0aa1